### PR TITLE
No more css and themes on the web views

### PR DESCRIPTION
**Pull request by sadasant**
These classes around CSS and Themes are not going to work on the web side, and we've confirmed over the chat that these are no longer needed. Please, let me know if I got that wrong.

How does this look?

### CODE REVIEW

**Review by rchiodo — APPROVED** [review]

**Review by DonJayamanne — CHANGES_REQUESTED** [review]
I think we still need to know on the UI side whether its light or dark theme, as that controls the icons in webviews.
Hence we cannot sending the value `isDark` to the UI.

File: src/webviews/extension-side/webviewHost.ts [hunk 1]
```diff
@@ -161,10 +159,6 @@ export abstract class WebviewHost<IMapping> implements IDisposable {
                 this.webViewRendered();
                 break;
 
-            case CssMessages.GetCssRequest:
```
Comment (DonJayamanne) [inline src/webviews/extension-side/webviewHost.ts:164]: I'd rename this to `GetTheme` which would return `isDark`

File: src/webviews/extension-side/webviewHost.ts [hunk 2]
```diff
@@ -296,23 +290,6 @@ export abstract class WebviewHost<IMapping> implements IDisposable {
...
-        this.setTheme(requestIsDark);
-        const isDark = settings.ignoreVscodeTheme
-            ? false
-            : await this.themeFinder.isThemeDark(settings.extraSettings.theme);
-        const resource = this.owningResource;
-        const css = await this.cssGenerator.generateThemeCss(resource, requestIsDark, settings.extraSettings.theme);
-        return this.postMessageInternal(CssMessages.GetCssResponse, {
-            css,
-            theme: settings.extraSettings.theme,
-            knownDark: isDark
```
[the remark this answers is not in the document]
Comment (DonJayamanne) [reply src/webviews/extension-side/webviewHost.ts:312]: Removing the CSS as is will actually break things, we need to ensure we don't set any CSS anywhere, and just use the styles injected by VS Code (removing our styles for font size & family will fix this).

**Comment by rchiodo** [comment]
> I think we still need to know on the UI side whether its light or dark theme, as that controls the icons in webviews. Hence we cannot sending the value `isDark` to the UI.

I thought 'isDark' requires reading the theme json though? Maybe we can determine it some other way.

File: src/webviews/extension-side/webviewHost.ts (hunk 1 above)
Comment (sadasant) [reply src/webviews/extension-side/webviewHost.ts:164]: Thank you! can do!

File: src/webviews/extension-side/webviewHost.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (sadasant) [reply src/webviews/extension-side/webviewHost.ts:312]: Thank you! can do!

**Comment by DonJayamanne** [comment]
> I thought 'isDark' requires reading the theme json though?

Nope

> Maybe we can determine it some other way.

Yes,
we can use the `activeColorTheme` property in `vscode.window`

File: src/webviews/extension-side/webviewHost.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (DonJayamanne) [reply src/webviews/extension-side/webviewHost.ts:312]: I think we should also remove the following, as this should already be setup by VS Code for webviews.
`background-color: var(--vscode-editor-background);`

[the remark this answers is not in the document]
Comment (DonJayamanne) [reply src/webviews/extension-side/webviewHost.ts:312]: We should also remove the file `cssCodeGenerator.node.ts` as thats not required (basically if we're removing the loading of themes from files then this file too needs to go)

[the remark this answers is not in the document]
Comment (DonJayamanne) [reply src/webviews/extension-side/webviewHost.ts:312]: Similarly remote `IThemeFinder` and `ICodeCssGenerator`

**Comment by sadasant** [comment]
@DonJayamanne thank you so much!! Your feedback helps me a lot 🙏

**Review by rchiodo — APPROVED** [review]

File: src/webviews/webview-side/interactive-common/common.css [hunk 3]
```diff
@@ -10,8 +10,8 @@ html {
 }
 
 #main-panel {
-    background: var(--override-background, var(--vscode-editor-background));
-    color: var(--override-foreground, var(--vscode-editor-foreground));
+    background: var(--vscode-editor-background);
+    color: var(--vscode-editor-foreground);
```
Comment (DonJayamanne) [inline src/webviews/webview-side/interactive-common/common.css:14]: Do we need this? I would have thought this is not necessary.

File: src/webviews/webview-side/interactive-common/common.css [hunk 4]
```diff
@@ -10,8 +10,8 @@ html {
 }
 
 #main-panel {
-    background: var(--override-background, var(--vscode-editor-background));
-    color: var(--override-foreground, var(--vscode-editor-foreground));
+    background: var(--vscode-editor-background);
```
Comment (DonJayamanne) [inline src/webviews/webview-side/interactive-common/common.css:13]: Do we need this? I would have thought this is not necessary.

Comment (DonJayamanne) [reply src/webviews/webview-side/interactive-common/common.css:13]: Doesn't matter

File: src/webviews/webview-side/interactive-common/common.css (hunk 3 above)
Comment (DonJayamanne) [reply src/webviews/webview-side/interactive-common/common.css:14]: Doestn' matter

Comment (sadasant) [reply src/webviews/webview-side/interactive-common/common.css:14]: Just trying to make the least changes possible, `--vscode-editor-foreground` is in many places 😅

File: src/webviews/webview-side/interactive-common/common.css (hunk 4 above)
Comment (sadasant) [reply src/webviews/webview-side/interactive-common/common.css:13]: Just trying to make the least changes possible, `--vscode-editor-foreground` is in many places 😅

**Pull request merged**